src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHost.cs

```diff
@@ -65,11 +65,18 @@ public static SynchronizationContext CurrentOrMainJSSynchronizationContext
             get
             {
 #if FEATURE_WASM_THREADS
-                return JSSynchronizationContext.CurrentJSSynchronizationContext ?? JSSynchronizationContext.MainJSSynchronizationContext!;
+                return (JSProxyContext.ExecutionContext ?? JSProxyContext.MainThreadContext).SynchronizationContext;
 #else
                 return null!;
```

Comment:
Why are we doing `null!` here btw? Shouldn't the return type just be `SynchronizationContext?`

Reply:
Doesn't matter, this code will go away in next PR.